### PR TITLE
enhance url composition in sonarr couchpotato and sickbeard plugins

### PR DESCRIPTION
**Pull request by midraal**
should fix the use case where these plugins are not located at the server root path or behind a http proxy

### CODE REVIEW

**Comment by cvium** [comment]
is this ready to be merged then?

**Comment by midraal** [comment]
yes, in manual testing it seems to work, but don't have an install wich is like @grrr2

**Comment by liiight** [comment]
I'm not sure this'll do the trick. In @grrr2 example, the user can set a URL that doesn't even include the port setting as it's behind a proxy http:///ur/sickbeard/api.
Hence, adding the port setting may not be required and can even break the URL (unless the user inputs the port as 80 which is redundant.

I think the best choice is to let to user to enter the entirety of the URL including the port number as a single parameter.

**Comment by midraal** [comment]
changing the default port to 80 should make the use case of being behind a proxy slightly more elegant

File: flexget/plugins/input/couchpotato.py [hunk 1]
```diff
@@ -15,7 +16,7 @@ class CouchPotato(object):
         'type': 'object',
         'properties': {
             'base_url': {'tpye': 'string'},
-            'port': {'type': 'number'},
+            'port': {'type': 'number', 'default': 80},
             'api_key': {'type': 'string'}
         },
         'required': ['api_key', 'base_url', 'port'],
```
Comment (liiight) [inline flexget/plugins/input/couchpotato.py]: you should remove the port from the required properties

Comment (midraal) [inline flexget/plugins/input/couchpotato.py]: can't believe i missed that

**Comment by liiight** [comment]
Any chance merging this? I want to add some stuff to those plugins as well

**Comment by cvium** [comment]
Well, did you agree on how to handle the various urls?

**Comment by liiight** [comment]
yeah, the proposed fix is the agreed upon implementation

**Comment by cvium** [comment]
There are some conflicts though because of a recent merge. If @midraal fixes it I'll merge it. I'd rather not fiddle around with the cli.

**Comment by midraal** [comment]
should be okay to merge now

**Pull request merged**